client/components/jetpack/card/jetpack-product-card/index.tsx

```diff
@@ -104,7 +113,7 @@ const DisplayPrice = ( {
 		);
 	}
 
-	if ( isFree ) {
+	if ( isFree && ! isJetpackFree ) {
```

Comment:
Checking the product slug in this component is not scalable, but given this is an experiment it's the cleanest way of doing for now. We can revisit this depending on the experiment outcome.